ovs/extensions/generic/heartbeat.py

```diff
@@ -50,21 +50,23 @@ def pulse():
                                             EtcdConfiguration.get('/ovs/framework/messagequeue|password'),
                                             EtcdConfiguration.get('/ovs/framework/hosts/{0}/ip'.format(machine_id)))
 
-        celery_path = OSManager.get_path('celery')
-        worker_states = check_output("{0} inspect ping -b {1} --timeout=5 2> /dev/null | grep OK | perl -pe 's/\x1b\[[0-9;]*m//g' || true".format(celery_path, amqp), shell=True)
+        celery = Celery(broker=amqp)
+        worker_states = celery.control.inspect().ping()
```

Comment:
Default timeout is 1 sec. [celery.app.control](http://docs.celeryproject.org/en/latest/reference/celery.app.control.html)